### PR TITLE
Adopt Docs Starter Pack config

### PR DESCRIPTION
**Pull request by rkratky**
Fixes [FR-5577](https://warthogs.atlassian.net/browse/FR-5577).

This puts in place the same Read the Docs and Sphinx config as used by the https://github.com/canonical/sphinx-docs-starter-pack, which will make maintenance and upgrades easier.

[FR-5577]: https://warthogs.atlassian.net/browse/FR-5577?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dbungert** [comment]
@rkratky - thanks for this!

I haven't reviewed the content too closely, just looking at CI as of yet, will look at content later.
The new test is failing, I'll let you consider what's going on there.

CLA check is also failing - https://github.com/canonical/subiquity/actions/runs/6422328325/job/17438594649?pr=1822#step:2:50 - I usually suggest using the `@canonical` email address, which should address this.

**Comment by rkratky** [comment]
Yeah. I forgot the CLA, but that's taken care of now. I'll need to look more closely at why the docs checks are failing.

File: doc/reuse/links.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.. _reStructuredText style guide: https://canonical-documentation-with-sphinx-and-readthedocscom.readthedocs-hosted.com/style-guide/
+.. _Read the Docs at Canonical: https://library.canonical.com/documentation/read-the-docs
+.. _How to publish documentation on Read the Docs: https://library.canonical.com/documentation/publish-on-read-the-docs
```
Comment (dbungert) [inline doc/reuse/links.txt:3]: I presume library.canonical.com is not accessible to community contributors, is that correct?

**Review by dbungert — COMMENTED** [review]
This looks mostly fine when the doccheck is fixed (I love the dead link checker), one question though.

**Review by Chris-Peterson444 — COMMENTED** [review]
This looks good! Thanks @rkratky 

Also just one question from me.

File: doc/custom_conf.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+#
+# For the full list of built-in configuration values, see the documentation:
+# https://www.sphinx-doc.org/en/master/usage/configuration.html
+
+############################################################
+### Project information
+############################################################
+
+# Product name
+project = 'Ubuntu Install Guide'
```
[the remark this answers is not in the document]
Comment (dbungert) [reply doc/custom_conf.py:18]: This was how we described the project.  The goal here is eventually an overall set of docs, not just ones focused almost exclusively on autoinstall.

File: doc/reuse/links.txt (hunk 1 above)
Comment (rkratky) [reply doc/reuse/links.txt:3]: Goog point. I'll remove those.

File: doc/custom_conf.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (rkratky) [reply doc/custom_conf.py:18]: Side note: once we're ready for publishing, I'd vote for `s/Install/Installation/`. Using "install" as a noun is not pretty.

**Comment by rkratky** [comment]
@dbungert Could you please approve now that the Subiquity checks have passed?

I'll need to work on the docs checks, but it will take a bit more time. Two of them are failing to run, so I need to find out why. The third one (linkcheck) is failing because of, well, invalid links -- so I'll submit another PR to fix them.

File: doc/custom_conf.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dbungert) [reply doc/custom_conf.py:18]: Yes, "Installation" sounds better.  Thanks.

**Comment by dbungert** [comment]
> I'll need to work on the docs checks, but it will take a bit more time. Two of them are failing to run, so I need to find out why. The third one (linkcheck) is failing because of, well, invalid links -- so I'll submit another PR to fix them.

I think the linkcheck links are failing on the manpages, can you confirm?  We were purposefully using links that are redirecting to the latest series, does linkcheck not like those?   Those links do work if I click them.

**Review by dbungert — APPROVED** [review]

**Pull request merged**